### PR TITLE
Validate the presence and contents of glyphs and sprite urls

### LINKED ISSUE #170: Are sprites, glyphs, and constants required?
2014-09-22 at 11 42 am

### LINKED ISSUE #299: validate presence of glyphs/sprite when text/icons are used
Validation should catch an issue like https://github.com/mapbox/mapbox-gl-js/issues/1291.

### PR DESCRIPTION
**Pull request by lucaswoj**
fixes #299
fixes #170 

cc @jfirebaugh @ansis @bhousel @tmcw @mourner

### CODE REVIEW

File: test/fixture/missing-glyphs.output.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[
+  {
+    "message": "layers[0].layout.text-field: \"glyphs\" url must be specified",
```
Comment (jfirebaugh) [inline test/fixture/missing-glyphs.output.json]: Can this be a little more descriptive?

> the use of "text-field" requires a style with a "glyphs" property

File: lib/validate/validate_glyphs_url.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var ValidationError = require('../error/validation_error');
+
+module.exports = function(options) {
```
Comment (tmcw) [inline lib/validate/validate_glyphs_url.js:6]: if glyphs is set to `true`, will this still run, or will that be caught ahead of time as a wrong type?

File: test/fixture/missing-glyphs.output.json (hunk 1 above)
Comment (lucaswoj) [inline test/fixture/missing-glyphs.output.json]: :+1: changed to 

> use of "icon-image" requires a stylesheet "sprite" property
> use of "text-field" requires a stylesheet "glyphs" property

File: lib/validate/validate_glyphs_url.js (hunk 2 above)
Comment (lucaswoj) [inline lib/validate/validate_glyphs_url.js:6]: Good catch. I added a call to `validateString` in this method.

File: test/fixture/missing-glyphs.output.json (hunk 1 above)
Comment (jfirebaugh) [inline test/fixture/missing-glyphs.output.json]: `s/stylesheet/style`

Comment (lucaswoj) [inline test/fixture/missing-glyphs.output.json]: Oops. That makes sense. My apologies.

**Pull request merged**